java/vector/src/main/java/org/apache/arrow/vector/BaseDataValueVector.java

```diff
@@ -30,6 +30,9 @@
 
   protected final static byte[] emptyByteArray = new byte[]{}; // Nullable vectors use this
 
+  /** maximum extra size at the end of the buffer */
+  private static final int MAX_BUFFER_PADDING = 0; // TODO: check
```

Comment:
Currently, I'm just enforcing that the buffer is exactly the size it should be.
What is the right value for this? 256 ?
Also I need to add tests that have a larger padded buffer.